pkg/submitter/submitter.go

```diff
@@ -0,0 +1,504 @@
+							level.Error(s.logger).Log("msg", "miner is not in a status that can submit", "statusID", statusID)
+							<-ticker.C
+							continue
+						}
+						s.blockUntilTimeToSubmit(s.ctx)
+						tx, err := s.Submit(s.ctx, result)
+						if err != nil {
+							s.submitFailCount.Inc()
+							level.Error(s.logger).Log("msg", "submiting a solution, retrying", "err", err, "account", s.account.Address.String())
+							<-ticker.C
+							continue
+						}
+						level.Debug(s.logger).Log("msg", "submited a solution", "txHash", tx.Hash().String(), "account", s.account.Address.String())
+						s.saveGasUsed(ctx, tx)
+						s.submitCount.Inc()
+						return
+					}
+				}
+			}
+		}
+	}(ctx, result)
+}
+
+func (s *Submitter) Submit(ctx context.Context, result *mining.Result) (*types.Transaction, error) {
+	challenge := result.Work.Challenge
+	nonce := result.Nonce
+	s.currentChallenge = challenge
+	s.currentNonce = nonce
+
+	// The submit contains values for 5 data IDs so add them here.
+	for i := 0; i < 5; i++ {
+		valKey := fmt.Sprintf("%s%d", db.QueriedValuePrefix, challenge.RequestIDs[i].Uint64())
+		m, err := s.proxy.BatchGet([]string{valKey})
+		if err != nil {
+			return nil, errors.Wrapf(err, "retrieve pricing data for current request id")
+		}
+		val := m[valKey]
+		var value *big.Int
+		if len(val) == 0 {
+			jsonFile, err := os.Open(s.cfg.ManualDataFile)
+			if err != nil {
+				return nil, errors.Wrapf(err, "manualData read Error")
+			}
+			defer jsonFile.Close()
+			byteValue, _ := ioutil.ReadAll(jsonFile)
+			var result map[string]map[string]uint
+			_ = json.Unmarshal([]byte(byteValue), &result)
+			_id := strconv.FormatUint(challenge.RequestIDs[i].Uint64(), 10)
+			val := result[_id]["VALUE"]
+			if val == 0 {
+				return nil, errors.Errorf("retrieve pricing data for current request id")
+			}
+			value = big.NewInt(int64(val))
+		} else {
+			value, err = hexutil.DecodeBig(string(val))
+			if err != nil {
+				if challenge.RequestIDs[i].Uint64() > tracker.MaxPSRID() {
+					level.Error(s.logger).Log(
+						"msg", "decoding price value prior to submitt ing solution",
+						"err", err,
+					)
+					if len(val) == 0 {
+						level.Error(s.logger).Log("msg", "0 value being submitted")
+						s.currentValues[i] = big.NewInt(0)
+					}
+					continue
+				}
+				return nil, errors.Errorf("no value in database,  reg id:%v", challenge.RequestIDs[i].Uint64())
+			}
+		}
+		s.currentValues[i] = value
+	}
+
+	// Submit the solution.
+	tx, err := s.submitter.Submit(ctx, s.proxy, "submitSolution", s.submit)
+	if err == nil {
+		return tx, nil
+	}
+	level.Error(s.logger).Log("msg", "submit solution", "pubkey", s.submitter.Address().String())
+	return nil, errors.New("submitting solution txn by any account")
+}
+
+func (s *Submitter) getMinerStatus() int64 {
+	// Check if the staked account is in dispute before sending a transaction.
+	v, dispute := s.IsInDispute()
```

Comment:
Don't think we even the IsInDispute function.
We just want to check that the miner is in a status where it can submit
It is not in d
```suggestion
	v, dispute := s.IsInDispute()
```

Reply:
Done